developers/wallets/querybyexample.md

```diff
@@ -10,7 +10,58 @@ Prerequisite: This document assumes completion of the
 
 Responding to requests for credentials from a verifier is a critical step in building out a complete wallet. There are multiple credential query formats, but today we're focusing on Verifiable Presentation Request's `QueryByExample` format.
 
-When integrating with the CHAPI Playground, the JSON sent to the Wallet is in the form of a Verifiable Presentation (VP) Request with one or more QueryByExamples. A QueryByExample contains a property named `credentialQuery.example` which contains the set of properties that the Verifiable Credential (VC) must match. The `@context` and `type` properties of a VC will always be present. Additional properties may be requested for more fine grained querying.
+When integrating with the CHAPI Playground, the JSON sent to the Wallet is in the form of a Verifiable Presentation (VP) Request with one or more QueryByExamples. The payload will be received from the `handleGetEvent` function:
+
+```js
+async function handleGetEvent() {
+  const event = await WebCredentialHandler.receiveCredentialEvent();
+  console.log('Get Credential Event:', event.type, event);
+
+  const vp = event.credentialRequestOptions.web.VerifiablePresentation;
```

Comment:
The value of `event.credentialRequestOptions.web.VerifiablePresentation` is a Verifiable Presentation Request (VPR), so we should use the variable name `vpr` instead to help reduce confusion. It is specifically not a `VerifiablePresentation`.

CHAPI extends the Credential Management API by defining a `WebCredential` that implements the `Credential` interface (note that `Credential` is not a `VerifiableCredential` but a WebIDL interface specific to the browser Credential Management API spec). This is the expected mechanism for extending that spec (other `Credential` types include `FederatedCredential`, `PasswordCredential`, etc.).

Each of these generates C++ (or whatever native code) when implemented directly in the browser, so the number of types of "Credentials" is bounded, unlike how it is unbounded in the Verifiable Credential ecosystem. Also, a `WebCredential` is not specific to the Verifiable Credentials ecosystem either, allowing for any kind of credential data to be handled by a `CredentialHandler`. Thus, a `WebCredential` has its own `dataType` and `data` (properties of the `WebCredential`). The former is always `VerifiablePresentation` when either getting or storing Verifiable Credentials, as these are transmitted via Verifiable Presentations -- and the Credential Management API's design is such that you "get" and "store" the same "type" of `Credential`. This means that you either "get" a `WebCredential` that has, as its data, a `VerifiablePresentation`, or you "store" one -- at that layer.

Again, in a somewhat awkward fashion, this means that the `credentialRequestOptions` (again, from the core Credential Management API spec) expresses query information for specific `Credential` types using the prefix for the extending types as property names: `{web: {...}, federated: {...}, password: {...}}`. Therefore, when using CHAPI, `web` is used -- and within `web`, the specific `dataType` is further used to continue this design pattern: `web: {VerifiablePresentation: {...}}`. That does not mean that the value of `credentialRequestOptions.web.VerifiablePresentation` is a Verifiable Presentation -- as this is not how the Credential Management API data model pattern works. It is, instead, whatever "options" are needed to be passed to find a `WebCredential` of `dataType`: `VerifiablePresentation`. These "options" are defined as a `VerifiablePresentationRequest`, so the value is a VPR.

I'll convert this into an issue and hopefully the background can be helpful in some way.